applications/nrf_desktop/src/modules/ble_qos.c

```diff
@@ -0,0 +1,689 @@
+/*
```

Comment:
The ble_qos module has dependencies to HID, USB and SERIAL and more. Is it hard to extract these dependencies? That will make it a lot easier to move out this module to subsys/bluetooth later.

Reply:
Overall this module is closely tied to nrf_desktop, as it not only deals with the HCI commands and events, but also nrf_desktop-specific events and configuration channel interaction.
The USB and SERIAL dependencies comes from the statistics printout via CDC ACM (this part can easily be removed).

The code that handles the information contained in hci_vs_evt_qos_conn_event_report_t is generic and can be easily extracted from nrf_desktop (chmap_filter.h and libchmapfilt.a).
There would be some work in setting up a more generic module that handles the interaction between the HCI VS events (higher priority, high rate) and the processing (lower priority, low rate).

The approach preferred by the desktop team is to add this feature as part of nrf_desktop application initially, gather feedback, and rather move it to subsys/bluetooth when it is considered more mature.